Unity/Assets/Scripts/SCHIZO/VFX/MatPassID.cs

```diff
@@ -0,0 +1,109 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SCHIZO.VFX;
+
+/// <summary>
+/// Effects that support blend modes are: color tint, and image overlay. 
+/// </summary>
+public enum BlendMode
+{
+    Add = 0,
+    Normal = 1,
+    Multiply = 2,
+    Screen = 3,
+    Substract = 4,
+}
+
+/// <summary>
+/// Class that holds information about <see cref ="Effects"/> effect, <see cref ="BlendMode"/> blend mode for supported effect or int ID of shader pass to render with.
+/// Material Properties can be set via <c>SetVector()</c>, <c>SetFloat()</c>, <c>SetColor()</c>, <c>SetTexture()</c> methods.
+/// </summary>
+public sealed class MatPassID
+{
+    public MatPassID(Effects effect) { _effect = effect; _passID = 0; }
+    public MatPassID(Effects effect, int passID) : this(effect) { _passID = passID; }
+    public MatPassID(Effects effect, BlendMode blendMode) : this(effect) { _passID = (int) blendMode; }
+
+    private int _passID;
+    private Effects _effect;
+
+    private Dictionary <int, Vector4> vectors;
+    public void SetVector(string name, Vector4 value)
+    {
+        int id = Shader.PropertyToID(name);
+
+        vectors ??= [];
+        if (vectors.ContainsKey(id)) { vectors[id] = value; }
+        else { vectors.Add(id, value); }

```

Comment:
```suggestion
        vectors[id] = value;
```

same for the other ones

Reply:
first time using Dictionaries, so I was looking at [ examples ](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?view=net-8.0#examples) and assumed it will throw if key does not exist 🤷